modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/workflow/mgt/WorkflowManagementTestCase.java

```diff
@@ -915,6 +980,59 @@ public void testUpdateRoleListOfUserOperation() {
         } catch (Exception e) {
             log.error("Error while clean up testDeleteUserOperation");
         }
+
+        try {
+            client.addAssociation(workflowId, "TestDeleteRoleAssociation", WorkflowConstants.DELETE_ROLE_EVENT,
+                    "boolean(1)");
+            usmClient.deleteRole(roleName1);
+        } catch (Exception e) {
+            Assert.fail("failed to add deleteUser workflow.");
+        }
+        try {
+            usmClient.updateRoleListOfUser(userName2, new String[0], new String[]{roleName1});
+            Assert.fail("Should throw an exception since there is a pending workflow for deleting username1.");
+        } catch (Exception e) {
+            //test pass
+        }
+        try {
+            AssociationDTO[] associations = client.listAssociationsForWorkflow(workflowId);
+            for (AssociationDTO association : associations) {
+                if ("TestDeleteRoleAssociation".equals(association.getAssociationName())) {
+                    associationId = association.getAssociationId();
+                    client.deleteAssociation(associationId);
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error while clean up testDeleteUserOperation");
+        }
+
+        try {
+            client.addAssociation(workflowId, "TestRenameRoleAssociation", WorkflowConstants.UPDATE_ROLE_NAME_EVENT,
+                    "boolean(1)");
+            usmClient.updateRoleName(roleName2, roleName3);
+        } catch (Exception e) {
+            Assert.fail("failed to add renameRole workflow.");
+        }
+        try {
+            usmClient.updateRoleListOfUser(userName2, new String[]{}, new String[]{roleName2});
+            Assert.fail("Should throw an exception since there is a pending workflow for deleting username1.");
+        } catch (Exception e) {
+            //test pass
```

Comment:
going through catch clause is the happy path. that's why it has kept empty. if exception is not thrown from line 1018, then the test need to fail, this has been evaluated in assert at line 1019